pulp_ansible/app/viewsets.py

```diff
@@ -290,6 +320,45 @@ def sync(self, request, pk):
         )
         return OperationPostponedResponse(result, request)
 
+    @extend_schema(
+        description="Trigger an asynchronous task to sign Ansible content.",
+        responses={202: AsyncOperationResponseSerializer},
+    )
+    @action(detail=True, methods=["post"], serializer_class=AnsibleRepositorySignatureSerializer)
+    def sign(self, request, pk):
+        """
+        Dispatches a sync task.
+        """
+        content_units = {}
+
+        repository = self.get_object()
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        signing_service = serializer.validated_data["signing_service"]
+        content = serializer.validated_data["content_units"]
+
+        if "*" in content:
+            content_units = ["*"]
+        else:
+            for url in content:
+                content_units[NamedModelViewSet.extract_pk(url)] = url
+            content_units_pks = set(content_units.keys())
```

Comment:
dict keys are sufficiently unique already...
```suggestion
            content_units_pks = list(content_units.keys())
```